### PR TITLE
integrating data management into stellar-sdk #80

### PR DESCRIPTION
**Pull request by berlinguyinca**
This should provide a basic implementation for the missing features, outlined in this post:

https://github.com/stellar/java-stellar-sdk/issues/80

It will also add another plugin to the gradle build lifecycle, to automatically push the build jar to the local maven repository, for easier integration with maven based projects.

### CODE REVIEW

**Comment by berlinguyinca** [comment]
dear all,

any change to get this merged, so we can remove our fork?

thanks!

**Comment by berlinguyinca** [comment]
dear all,

would it be possible to get an update too this pull requests?

thanks!

**Comment by berlinguyinca** [comment]
hello, could you please get an update to this? It's been over a month now

File: build.gradle [hunk 1]
```diff
@@ -20,11 +21,24 @@ publishing {
...
+            groupId 'org.stellar.sdk'
+            artifactId 'stellar-sdk'
+            from components.java
+        }
     }
     repositories {
         ivy {
             url "$projectDir/repos"
         }
+	mavenLocal()
```
Comment (bartekn) [inline build.gradle]: Can you remove changes to this file? It's not in a scope of this PR.

File: src/main/java/org/stellar/sdk/responses/GsonSingleton.java [hunk 2]
```diff
@@ -1,14 +1,16 @@
...
+import com.google.gson.*;
 import com.google.gson.reflect.TypeToken;
 
 import org.stellar.sdk.Asset;
 import org.stellar.sdk.KeyPair;
 import org.stellar.sdk.responses.effects.EffectResponse;
 import org.stellar.sdk.responses.operations.OperationResponse;
 
+import java.lang.reflect.Type;
+import java.util.Base64;
```
Comment (bartekn) [inline src/main/java/org/stellar/sdk/responses/GsonSingleton.java]: AFAIR, this package does not work in Android. Please use `com.google.common.io.BaseEncoding`.

File: src/main/java/org/stellar/sdk/responses/GsonSingleton.java [hunk 3]
```diff
@@ -43,9 +45,24 @@ public static Gson getInstance() {
                       .registerTypeAdapter(tradePageType.getType(), new PageDeserializer<TradeResponse>(tradePageType))
                       .registerTypeAdapter(tradeAggregationPageType.getType(), new PageDeserializer<TradeAggregationResponse>(tradeAggregationPageType))
                       .registerTypeAdapter(transactionPageType.getType(), new PageDeserializer<TransactionResponse>(transactionPageType))
+                      .registerTypeHierarchyAdapter(byte[].class,new ByteArrayToBase64TypeAdapter())
```
Comment (bartekn) [inline src/main/java/org/stellar/sdk/responses/GsonSingleton.java]: From Gson [docs](https://google.github.io/gson/apidocs/com/google/gson/GsonBuilder.html#registerTypeHierarchyAdapter-java.lang.Class-java.lang.Object-):
> Configures Gson for custom serialization or deserialization for an inheritance type hierarchy.

Is this the case here. Wouldn't _normal_ `registerTypeAdapter` work here?

Comment (berlinguyinca) [reply src/main/java/org/stellar/sdk/responses/GsonSingleton.java]: not sure, just wanted to have a working solution

File: src/main/java/org/stellar/sdk/responses/GsonSingleton.java (hunk 2 above)
Comment (berlinguyinca) [reply src/main/java/org/stellar/sdk/responses/GsonSingleton.java]: ok I look into it

File: build.gradle (hunk 1 above)
Comment (berlinguyinca) [reply build.gradle]: will do

**Comment by bartekn** [comment]
@berlinguyinca do you want to continue working on this?

**Pull request closed without merging**